src/renderer/renderScene.ts

```diff
@@ -51,28 +66,22 @@ export function renderScene(
   }
   context.fillStyle = fillStyle;
 
-  sceneState = {
-    ...sceneState,
-    scrollX: typeof offsetX === "number" ? offsetX : sceneState.scrollX,
-    scrollY: typeof offsetY === "number" ? offsetY : sceneState.scrollY,
-  };
+  // Fake zoom by scaling elements
+  const scaledElements = elements.map(element =>
+    getScaledElement(element, sceneState.zoom),
+  );
 
-  let atLeastOneVisibleElement = false;
-  elements.forEach(element => {
-    if (
-      !isVisibleElement(
-        element,
-        sceneState.scrollX,
-        sceneState.scrollY,
-        // If canvas is scaled for high pixelDeviceRatio width and height
-        // setted in the `style` attribute
-        parseInt(canvas.style.width) || canvas.width,
```

Comment:
Using `canvas.style.width` and `canvas.style.height` is a mistake I believe, on retina mobile elements would disappear even though they should still be visible.

Reply:
Yeah, definitely shouldn’t